### PR TITLE
[core] feat: CompoundTag

### PR DESCRIPTION
**Pull request by adidahiya**
#### Checklist

- [x] Includes tests
- [x] Update documentation

#### Changes proposed in this pull request:

New **CompoundTag** component

#### Reviewers should focus on:

Implementation matches Figma designs

#### Screenshot

| Variant | Light theme | Dark theme |
| - | - | - |
| Default | image | image |
| Intent | image | image |
| Minimal | image | image |
| Minimal intent | image | image |

| Round | Large | Large + Fill + Icons |
| - | - | - |
| image | image | image |

Interactive demo

Screen Recording 2024-01-30 at 12 14 08 PM

### CODE REVIEW

**Comment by adidahiya** [comment]
<h3>Add unit tests</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/2b9c4299-7c43-4ae7-a56b-689cc3a91fd2/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/2b9c4299-7c43-4ae7-a56b-689cc3a91fd2/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/2b9c4299-7c43-4ae7-a56b-689cc3a91fd2/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/2b9c4299-7c43-4ae7-a56b-689cc3a91fd2/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

File: packages/core/src/common/_variables-extended.scss [hunk 1]
```diff
@@ -11,7 +11,12 @@
 
 $half-grid-size: $pt-grid-size * 0.5 !default;
 
-// Extended color pallete
+// Extended color palette
+$gray0: #535d6a !default;
+$blue0: #11376b !default;
+$green0: #0f3e25 !default;
+$orange0: #5f380b !default;
+$red0: #782326 !default;
```
Comment (adidahiya) [inline packages/core/src/common/_variables-extended.scss]: @jessperrin @CPerinet I made up some color values here; I needed some colors 1 shade darker than `$gray1`, `$blue1`, etc.

**Comment by jessperrin** [comment]
These are the values for the minimal tag secondary background: 
image

And the values for default tag secondary background:
image

**Comment by jessperrin** [comment]
One more update: Per the darkmode updates, darkmode default tag colors should be the same as lightmode default tag, both backgrounds + text

**Comment by adidahiya** [comment]
<h3>fix tag tests</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/74d3f5e8-d6c7-4d1f-8e9f-cb66ef635959/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/74d3f5e8-d6c7-4d1f-8e9f-cb66ef635959/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/74d3f5e8-d6c7-4d1f-8e9f-cb66ef635959/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/74d3f5e8-d6c7-4d1f-8e9f-cb66ef635959/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

**Comment by adidahiya** [comment]
@jessperrin thanks for the color table, I will fix those.

The hover and active colors are under-specified right now -- I am using the darker values (referenced in my PR comment above) for active states like this:

image

**Comment by adidahiya** [comment]
<h3>fix colors</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/d5f17378-174e-4f0d-9015-eac6ea3a9c4a/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/d5f17378-174e-4f0d-9015-eac6ea3a9c4a/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/d5f17378-174e-4f0d-9015-eac6ea3a9c4a/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/d5f17378-174e-4f0d-9015-eac6ea3a9c4a/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

**Comment by jessperrin** [comment]
Played around with the docs, a couple notes: 
-The non-rounded compound tag should still have the same outside corner radius as the normal tag (2px?), instead of the square corners
-Adding the left icon but not the right increases the left tag height to be mismatched with the right: 
image

**Comment by adidahiya** [comment]
<h3>remove multiline prop from CompoundTag</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/56c118e1-fdcb-45ee-9d56-c982e1c06531/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/56c118e1-fdcb-45ee-9d56-c982e1c06531/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/56c118e1-fdcb-45ee-9d56-c982e1c06531/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/56c118e1-fdcb-45ee-9d56-c982e1c06531/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

**Comment by adidahiya** [comment]
<h3>fix tests, make leftContent and children required</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/fff31431-1ad2-4507-b149-9c30b6429a3f/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/fff31431-1ad2-4507-b149-9c30b6429a3f/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/fff31431-1ad2-4507-b149-9c30b6429a3f/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/fff31431-1ad2-4507-b149-9c30b6429a3f/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

**Pull request merged**